sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRegistryClientTestBase.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.containers.containerregistry;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.test.TestBase;
+
+import com.azure.core.test.TestMode;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.azure.containers.containerregistry.TestUtils.DEFAULT_POLL_INTERVAL;
+import static com.azure.containers.containerregistry.TestUtils.ONE_NANO_DURATION;
+
+public class ContainerRegistryClientTestBase extends TestBase {
+
+    private static final String AZURE_FORM_RECOGNIZER_ENDPOINT = "CONTAINERREGISTRY_ENDPOINT";
+    private static final String INVALID_KEY = "invalid key";
+
+    private static final Pattern JSON_PROPERTY_VALUE_REDACTION_PATTERN
+        = Pattern.compile("(\".*_token\":\"(.*)\".*)");
+
+    private Duration durationTestMode;
+    @Override
+    protected void beforeTest() {
+        if (interceptorManager.isPlaybackMode()) {
+            durationTestMode = ONE_NANO_DURATION;
+        } else {
+            durationTestMode = DEFAULT_POLL_INTERVAL;
+        }
+    }
+
+    ContainerRegistryClientBuilder getContainerRegistryBuilder(HttpClient httpClient) {
+        Configuration configuration = new Configuration()
+            .put("java.net.useSystemProxies", "true")
+            .put("http.proxyHost", "localhost")
+            .put("http.proxyPort", "8888")
+            .put("http.proxyUser", "1")
+            .put("http.proxyPassword", "1");
```

Comment:
You'll want to remove this before merging as live tests will fail. I've seen something similar to this in multiple project, mind filing an issue to add infrastructure to `azure-core-test` which allows enables test proxying when an environment variable is set.

Reply:
I actually could not get proxy to work because it needs to be passed to the httpBuilder. So, yeah azure-core-test will be the right place to have it - [https://github.com/Azure/azure-sdk-for-java/issues/20184](https://github.com/Azure/azure-sdk-for-java/issues/20184)

---
In reply to: [602633935](https://github.com/Azure/azure-sdk-for-java/pull/20139#discussion_r602633935) [](ancestors = 602633935)